tests/Strictness.hs

```diff
@@ -0,0 +1,42 @@
+-- Strictness tests.
+
+{-# LANGUAGE CPP, TemplateHaskell #-}
+import Test.QuickCheck
+
+import Control.Exception (Exception (..), throw)
+
+#if MIN_VERSION_containers(0,5,0)
```

Comment:
I'm inclined to say we don't need these if we have the right GHC version bound in the `cabal` file.

Reply:
You mean, don't build the module at all? We could do that, but it means using the GHC version as a conditional rather than the more precise and meaningful conditional on the version of `containers`. It's also guaranteed to match the conditional on the feature that's being tested.

It would be possible to add a Cabal flag that the solver can set automatically via a bound on `containers`, but I feel it's better not to add to the complexity of the Cabal file.

Reply:
That sounds like too much work for too little gain. Let's keep it like this.